marker_maze.py

```diff
@@ -0,0 +1,242 @@
+from img_base_class import *
+import cv2.aruco as aruco
+
+# Image stream processing thread
+class StreamProcessor(threading.Thread):
```

Comment:
This is somewhat common with the stuff in `rainbow.py`, isn't it? Perhaps we can refactor to have a base class for the `StreamProcessor`.. 

this is fine, but we should at least raise an issue to address the code duplication.

Reply:
yes, this is as we discussed. issue raised